File: binding/src/multimedia/libqvideoframeinput.cpp
#include <QMediaCaptureSession>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QVideoFrame>
#include <QVideoFrameFormat>
#include <QVideoFrameInput>
#include <qvideoframeinput.h>
#include "libqvideoframeinput.h"
#include "libqvideoframeinput.hxx"

QVideoFrameInput* QVideoFrameInput_new() {
	 return new QVideoFrameInput();
}

QVideoFrameInput* QVideoFrameInput_new2(const QVideoFrameFormat* format) {
	 return new QVideoFrameInput(*format);
}

QVideoFrameInput* QVideoFrameInput_new3(QObject* parent) {
	 return new QVideoFrameInput(parent);
}

QVideoFrameInput* QVideoFrameInput_new4(const QVideoFrameFormat* format, QObject* parent) {
	 return new QVideoFrameInput(*format, parent);
}

libqt_string QVideoFrameInput_Tr(const char* s) {
	QString _ret = QVideoFrameInput::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QVideoFrameInput_SendVideoFrame(QVideoFrameInput* self, const QVideoFrame* frame) {
	return self->sendVideoFrame(*frame);
}

QVideoFrameFormat* QVideoFrameInput_Format(const QVideoFrameInput* self) {
	return new QVideoFrameFormat(self->format());
}

QMediaCaptureSession* QVideoFrameInput_CaptureSession(const QVideoFrameInput* self) {
	return self->captureSession();
}

void QVideoFrameInput_ReadyToSendVideoFrame(QVideoFrameInput* self) {
	self->readyToSendVideoFrame();
}

void QVideoFrameInput_Connect_ReadyToSendVideoFrame(QVideoFrameInput* self, intptr_t slot) {
    void (*slotFunc)(QVideoFrameInput*) = reinterpret_cast<void (*)(QVideoFrameInput*)>(slot);
    QVideoFrameInput::connect(self, &QVideoFrameInput::readyToSendVideoFrame, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QVideoFrameInput_Tr2(const char* s, const char* c) {
	QString _ret = QVideoFrameInput::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QVideoFrameInput_Tr3(const char* s, const char* c, int n) {
	QString _ret = QVideoFrameInput::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QVideoFrameInput_Delete(QVideoFrameInput* self) {
    delete self;
}

File: binding/include/multimedia/libqvideoframeinput.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qvideoframeinput.html
pub const qvideoframeinput = struct {

    /// New constructs a new QVideoFrameInput object.
    pub fn New() QtC.QVideoFrameInput {
        return qtc.QVideoFrameInput_new();
    }


    /// New2 constructs a new QVideoFrameInput object.
    pub fn New2(format: ?*anyopaque) QtC.QVideoFrameInput {
        return qtc.QVideoFrameInput_new2(@ptrCast(format));
    }


    /// New3 constructs a new QVideoFrameInput object.
    pub fn New3(parent: ?*anyopaque) QtC.QVideoFrameInput {
        return qtc.QVideoFrameInput_new3(@ptrCast(parent));
    }


    /// New4 constructs a new QVideoFrameInput object.
    pub fn New4(format: ?*anyopaque, parent: ?*anyopaque) QtC.QVideoFrameInput {
        return qtc.QVideoFrameInput_new4(@ptrCast(format), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVideoFrameInput_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvideoframeinput.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SendVideoFrame(self: ?*anyopaque, frame: ?*anyopaque) bool {
        return qtc.QVideoFrameInput_SendVideoFrame(@ptrCast(self), @ptrCast(frame));
    }

    pub fn Format(self: ?*anyopaque, ) QtC.QVideoFrameFormat {
        return qtc.QVideoFrameInput_Format(@ptrCast(self));
    }

    pub fn CaptureSession(self: ?*anyopaque, ) QtC.QMediaCaptureSession {
        return qtc.QVideoFrameInput_CaptureSession(@ptrCast(self));
    }

    pub fn ReadyToSendVideoFrame(self: ?*anyopaque, ) void {
        qtc.QVideoFrameInput_ReadyToSendVideoFrame(@ptrCast(self));
    }

    pub fn OnReadyToSendVideoFrame(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QVideoFrameInput_Connect_ReadyToSendVideoFrame(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVideoFrameInput_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvideoframeinput.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QVideoFrameInput_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qvideoframeinput.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QVideoFrameInput_Delete(@ptrCast(self));
    }
};
